testExternalSiteRedirect fails due to twitter.com returning 403 instead of expected 200

**Jetty version(s)**
Jetty 12.0.x

**Jetty Environment**
core

**Java version/vendor** `(use: java -version)`
openjdk version "21.0.8" 2025-07-15
OpenJDK Runtime Environment Homebrew (build 21.0.8)
OpenJDK 64-Bit Server VM Homebrew (build 21.0.8, mixed mode, sharing)

**OS type/version**
macOS 15.5 (24F74)

**Description**
Good evening, the testExternalSiteRedirect test fails, when I run "mvn install", because the URL https://twitter.com/twitter returns 403 instead of 302 (and faking a User-Agent does not change that):

```
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.991 s <<< FAILURE! -- in org.eclipse.jetty.client.ExternalSiteTest
[ERROR] org.eclipse.jetty.client.ExternalSiteTest.testExternalSiteRedirect -- Time elapsed: 0.974 s <<< FAILURE!
org.opentest4j.AssertionFailedError: expected: <200> but was: <400>
	at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
	at org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
	at org.junit.jupiter.api.AssertEquals.failNotEqual(AssertEquals.java:197)
	at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:150)
	at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:145)
	at org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:531)
	at org.eclipse.jetty.client@12.0.25-SNAPSHOT/org.eclipse.jetty.client.ExternalSiteTest.testExternalSiteRedirect(ExternalSiteTest.java:146)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   ExternalSiteTest.testExternalSiteRedirect:146 expected: <200> but was: <400>
```

The issue is not specific just to my Macbook Air M1:

`curl -I -L "https://twitter.com/twitter" 2>/dev/null | head -1` return 403

**How to reproduce?**

```
cd jetty-core/jetty-client && mvn clean test -Dtest=ExternalSiteTest#testExternalSiteRedirect -Dmaven.build.cache.enabled=false
```

Here are the logs (please wait for my upcoming PR):

[testExternalSiteRedirect-eclipse.txt](https://github.com/user-attachments/files/21700861/testExternalSiteRedirect-eclipse.txt)
[testExternalSiteRedirect-twitter.txt](https://github.com/user-attachments/files/21700862/testExternalSiteRedirect-twitter.txt)

--- a/jetty-core/jetty-client/src/test/java/org/eclipse/jetty/client/ExternalSiteTest.java
+++ b/jetty-core/jetty-client/src/test/java/org/eclipse/jetty/client/ExternalSiteTest.java
@@ -132,7 +132,7 @@ public class ExternalSiteTest
     @Test
     public void testExternalSiteRedirect() throws Exception
     {
-        String host = "twitter.com";
+        String host = "eclipse.org";
         int port = 443;
 
         // Verify that we have connectivity
@@ -140,7 +140,7 @@ public class ExternalSiteTest
 
         ContentResponse response = client.newRequest(host, port)
             .scheme(HttpScheme.HTTPS.asString())
-            .path("/twitter")
+            .path("/jetty")
             .timeout(15, TimeUnit.SECONDS)
             .send();
         assertEquals(200, response.getStatus());

